src/Compilers/Core/Portable/Text/TextChangeRangeExtensions.cs

```diff
@@ -25,7 +24,9 @@ internal static class TextChangeRangeExtensions
             // we could apply each one individually like we do in SyntaxDiff::ComputeSpansInNew by calculating delta
             // between each change in changesInNextVersion which is already sorted in its textual position ascending order.
             // but end result will be same as just applying it once with encompassed text change range.
-            var newChange = TextChangeRange.Collapse(changesInNextVersion);
+            var newChange = changesInNextVersion.Count == 1
+                ? changesInNextVersion[0]
+                : TextChangeRange.Collapse(changesInNextVersion);
```

Comment:
Should this optimization be moved inside `TextChangeRange.Collapse` itself?

Reply:
I considered that, but that method has several usages in which the caller only has an IEnumerable. I instead chose to limit the changes to the caller that was showing up in the performance profile I was looking at.